lightning/src/ln/channel.rs

```diff
@@ -865,6 +870,13 @@ impl<Signer: Sign> Channel<Signer> {
 			return Err(APIError::APIMisuseError { err: format!("Holder selected channel  reserve below implemention limit dust_limit_satoshis {}", holder_selected_channel_reserve_satoshis) });
 		}
 
+		// Check that config has valid values set
+		if !Self::config_htlc_in_flight_value_is_valid(config.channel_options){
```

Comment:
Lets just bound the value instead of failing here, no?

Reply:
Do you mean that if the user sets a value that's `< 1`,  we'll set the `holder_max_htlc_value_in_flight_msat` using `1%` when creating the channel (and `100%` if `> 100`)?

Or do you mean that we should strictly bind the `ChannelHandshakeConfig` to only allow a `max_inbound_htlc_value_in_flight_percent_of_channel` value between 1-100? 

A strict binding would be a much nicer solution if there is an easy way to do that in Rust, but as I'm still quite new to Rust, I'm not aware an easy solution/type for that, and my searches yielded no great results. I was also a bit afraid if such a solution would create any implications for the language bindings, which is why I opted for dealing with it in the `Channel` struct instead. If you have any tips of how to solve this though, I'd be very thankful for those!

Reply:
> Do you mean that if the user sets a value that's < 1, we should set the holder_max_htlc_value_in_flight_msat using 1% when creating the channel (and 100% if > 100)?

This. I think the "most" rust-y way to do this would be to check the value when constructing the `ChannelHandshakeConfig`, but I dunno that we really want to go that far here - there's value, at least in the bindings, to just having a POD struct here. The docs should be really clear that it needs to be between 1-100, but if users set it to something crazy we can just bound it for them and use 1 or 100.